### ISSUE TITLE
New custom android component which uses TouchDelegate API

### ISSUE THREAD

**Issue by neha-b2001** [mozilla-mobile/fenix#19869]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have exactly one feature request described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Feature Requests are better when they’re open-ended instead of demanding a specific solution e.g: “I want an easier way to do X” instead of “add Y”)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

### What is the user problem or growth opportunity you want to see solved?
1. There are some buttons/clickable items for which Google Accessibility Scanner suggests **an increase in touch target size** to enhance accessibility. For example, the search bar - 
![image](https://user-images.githubusercontent.com/67039214/121003149-d19c9600-c7aa-11eb-990d-dd100cef51bf.png)

2. However, it is **not always possible to increase the size** of these clickable items, as it might disrupt the entire layout (especially when it is an important clickable item like the search bar). Even padding/margin alterations might end up disrupting rest of the layout. 
3. So, a clickable item with the **same size but larger touch area** is ideal, as mentioned in this comment https://github.com/mozilla-mobile/fenix/issues/16829#issuecomment-801464840_.
### How do you know that this problem exists today? Why is this important?
This problem has already come up in some issues -
1. https://github.com/mozilla-mobile/fenix/issues/16829 (TouchDelegate was used in a PR associated with this https://github.com/mozilla-mobile/fenix/pull/19676)
2. https://github.com/mozilla-mobile/fenix/issues/18765
3. https://github.com/mozilla-mobile/fenix/issues/9765

Fixing this issue is important as some buttons (like the cancel and microphone ones of #9765) need a larger touch area to enable easier clicking, but don't necessarily need larger size.  
### Who will benefit from it?
Any individual who happens to need buttons with a larger clickable area. It will help improve overall app accessibility. 
### Feature Request
1. The [TouchDelegate API](https://developer.android.com/reference/android/view/TouchDelegate) can be used to increase touch area of a button without resizing. (Thanks to this comment for mentioning this --> https://github.com/mozilla-mobile/fenix/issues/16829#issuecomment-825521395)
2. However, to use the TouchDelegate API for a button/clickable item, **this class should be used by an ancestor of the delegate** (delegate view is view whose touch area's to be changed).
3. Also, as far as I could make out from various articles/usage examples and trying this out myself, **an ancestor which uses TouchDelegate can only handle **one** child button's touch events**. So, for each clickable item whose touch area we want to increase, we might need to define some immediate ancestor (say, a RelativeLayout) with which TouchDelegate class can be used. (We can explore this further to see if there's some way of having multiple delegates for a single parent.)
4. It might be good to have a **custom Android component** to fulfil this need - a button or navigation bar with a certain size, wrapped in a RecyclerView (or some other custom view) which implements TouchDelegate. This way, for any project, user can increase the touch area for a button without altering its intended size with ease. 

[Will also have to check/ensure that TouchDelegate API works properly under all conditions] 

### Useful Links for TouchDelegate API
1. https://developer.android.com/training/gestures/viewgroup#delegate
2. https://medium.com/android-news/android-change-touch-area-of-view-by-touchdelegate-fc19f2a34021
3. https://cyrilmottier.com/2012/02/16/listview-tips-tricks-5-enlarged-touchable-areas/ (might help with multiple delegate views under a parent view)
4. https://gist.github.com/kaiwinter/b277d4ccc2dfb3c15eb6 (might help with multiple delegate views under a parent view)
5. https://github.com/brendanw/Touch-Delegates/blob/master/src/com/brendan/touchdelegates/MyTouchDelegate.java

**Comment by neha-b2001** [comment]
Also tagging @eliserichards 😄

**Comment by buttercookie42** [comment]
Note that TouchDelegate on older Android versions (possibly everything older than Pie?) suffers from a bug that might have to be worked around:
https://issuetracker.google.com/issues/65295909
https://github.com/olegcherr/TouchDelegate-Bug

We also used to have our own workaround with Fennec, see [bug 1394879](https://bugzilla.mozilla.org/show_bug.cgi?id=1394879) for some pointers to that code.

**Comment by buttercookie42** [comment]
… and we also invented [our own HoverDelegate](https://bugzilla.mozilla.org/show_bug.cgi?id=1396076) to handle hover events used for Talkback touch exploration mode, too.
